shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/lock/ZookeeperDistributeLock.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.data.pipeline.core.lock;
+
+import com.google.common.collect.Maps;
+import org.apache.shardingsphere.data.pipeline.core.context.PipelineContext;
+import org.apache.shardingsphere.mode.manager.cluster.coordinator.lock.service.LockRegistryService;
+import org.apache.shardingsphere.mode.repository.cluster.ClusterPersistRepository;
+import org.apache.shardingsphere.mode.repository.cluster.ClusterPersistRepositoryConfiguration;
+import org.apache.shardingsphere.spi.typed.TypedSPIRegistry;
+
+import java.util.Map;
+
+/**
+ * Zookeeper Distribute Lock.
+ */
+public final class ZookeeperDistributeLock {
+    
+    private static final LockRegistryService LOCK_REGISTRY_SERVICE;
+    
+    private static final Map<String, Boolean> LOCK_NAME_LOCKED_MAP;
+    
+    static {
+        ClusterPersistRepositoryConfiguration repositoryConfig = (ClusterPersistRepositoryConfiguration) PipelineContext.getModeConfig().getRepository();
+        ClusterPersistRepository repository = TypedSPIRegistry.getRegisteredService(ClusterPersistRepository.class, repositoryConfig.getType(), repositoryConfig.getProps());
+        repository.init(repositoryConfig);
+        LOCK_REGISTRY_SERVICE = new LockRegistryService(repository);
+        LOCK_NAME_LOCKED_MAP = Maps.newConcurrentMap();
+    }
```

Comment:
1. Do not init repository here, since it's referenced in `RuleAlteredJobWorker` and then in `BootstrapInitializer`, if initialization throw exception, then it'll cause proxy startup failure.

2. It's better to reuse existing repository but not create another one. Try `PipelineContext.getContextManager().getMetaDataContexts().getMetaDataPersistService().getPersistRepository()`, need to verify whether context and service is null or not.